File: backend-ms1/src/main/java/ma/maneo/khola/service/facade/admin/collaborateur/TechnicienAdminService.java
package ma.maneo.khola.service.facade.admin.collaborateur;

import java.util.List;
import ma.maneo.khola.bean.core.collaborateur.Technicien;
import ma.maneo.khola.dao.criteria.core.collaborateur.TechnicienCriteria;
import ma.maneo.khola.zynerator.service.IService;



public interface TechnicienAdminService extends  IService<Technicien,TechnicienCriteria>  {
    Technicien findByUsername(String username);
    boolean changePassword(String username, String newPassword);




}

File: frontend/src/app/controller/service/admin/collaborateur/TechnicienAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {TechnicienDto} from 'src/app/controller/model/collaborateur/Technicien.model';
import {TechnicienCriteria} from 'src/app/controller/criteria/collaborateur/TechnicienCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class TechnicienAdminService extends AbstractService<TechnicienDto, TechnicienCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/technicien/');
    }

    public constrcutDto(): TechnicienDto {
        return new TechnicienDto();
    }

    public constrcutCriteria(): TechnicienCriteria {
        return new TechnicienCriteria();
    }
}
